hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -1005,8 +1006,8 @@ static void validateSecondaryIndexSchemaEvolution(
     for (Map.Entry<String, String> entry : columnToIndexName.entrySet()) {
       String columnName = entry.getKey();
       String indexName = entry.getValue();
-      
-      Schema.Field tableField = tableSchema.getField(columnName);
+
+      HoodieSchemaField tableField = tableSchema.getField(columnName).get();
       
       if (tableField == null) {
```

Comment:
Let's update this to first return an `Option` and then we can check if it is present instead of checking for `null` here.